app/ui/src/app/integration/integration-actions-provider.service.ts

```diff
@@ -222,10 +222,10 @@ export class IntegrationActionsProviderService extends IntegrationActionsService
 
   replaceDraftAction(integration: Integration | IntegrationOverview, deployment: IntegrationDeployment | DeploymentOverview): Promise<any> {
       return this.integrationSupportService.getDeployment(integration.id, deployment.version.toString())
-        .map(_deployment => {
-          this.store.patch(<any>integration, {
+        .switchMap(_deployment => {
+          return this.store.patch(_deployment.integrationId, {
             steps: _deployment.spec.steps,
-            draft: true
+            isDraft: true
```

Comment:
Turns out `isDraft` is not persisted, I think I have a fix for this on the server. @gashcrumb  do you mind if I push to this PR. I've tested a bit it seems to function. Though I'm a bit lost on the whole meaning of _Published_. Seems that if I create a new draft with this change then the status switches to _Unpublished_. Is this status meant to show only the latest versions state?